internal/providers/factory.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2024 Stacklok, Inc.
```

Comment:
In a future PR, I'll write unit tests for these methods. Right now - their behaviour is validated through pre-existing tests which exercise these methods.